LiteCore/Query/QueryParser.cc

```diff
@@ -1346,7 +1346,9 @@ namespace litecore {
         if (_propertiesUseSourcePrefix && !property.empty()) {
             // Interpret the first component of the property as a db alias:
             require(property[0].isKey(), "Property path can't start with array index");
-            if (_aliases.size() > 1 || alias == _dbAlias) {
+            if (count_if(_aliases.begin(), _aliases.end(), [](decltype(_aliases)::value_type it) {
+                return  it.second != kResultAlias;
+            }) > 1 || alias == _dbAlias) {
```

Comment:
Ideally this would have the "simple check" first so it becomes ever so slightly faster.  Also if a const ref is allowed for the "it" type that would be preferable (`const auto&` might work too)

Reply:
Nice use of count_if, but the multi-line lambda inside an outer `if` feels too complex IMHO. I think it'd be clearer to break it up, with one statement assigning the `count_if` call to a variable, and then the `if` testing it.

And I agree with Jim's point, though right here it's less important since the query parser is not performance sensitive (though the query _runtime_ is.)

Reply:
Good suggestion. Will refactor it.